drivers/console/ipm_console_receiver.c

```diff
@@ -66,51 +69,29 @@ static void ipm_console_thread(void *arg1, void *arg2, void *arg3)
 			++pos;
 		}
 
-		/* ISR may have disabled the channel due to full buffer at
-		 * some point. If that happened and there is now room,
-		 * re-enable it.
```

Comment:
I don't understand why it's OK to remove this.  That condition (an interrupt delivering data that fills the buffer) seems like it can still happen.

Reply:
I don't want to use the flags(enabled and channel_disable), there is possibility that the flags are accessed by ISR and receiver thread at the same time if SMP enabled.